pkg/synthetictests/duplicated_events.go

```diff
@@ -611,6 +638,12 @@ func isOauthApiserverProbeErrorConnectionRefusedFailed(monitorEvent monitorapi.E
 	return isOperatorMatchRegexMessage(monitorEvent, "openshift-oauth-apiserver", regExp), nil
 }
 
+// isConnectionRefusedOnSingleNode returns true if the event matched has a connection refused message for single node events and is with in threshold.
+func isConnectionRefusedOnSingleNode(monitorEvent monitorapi.EventInterval, kubeClientConfig *rest.Config, count int) (bool, error) {
+	regExp := regexp.MustCompile(singleNodeErrorConnectionRefusedRegExpStr)
+	return isSingleNodeCluster(kubeClientConfig) && regExp.MatchString(monitorEvent.String()) && count < duplicateSingleNodeEventThreshold, nil
```

Comment:
This looks like we might be querying the Infrastructure object for every event interval, probably tens of thousands of times. Is there some precedent for a global check here in the code already for SNO?

Reply:
That's a good call out, I had assumed it was okay since all those allowed functions take kubeConfig as an arg, but it seems we already collect the topology info anyway, updated the code to make use of it. What do you think of the change?

Reply:
Looks good, sounds like trouble if anyone is actually using that kubeconfig elsewhere.